osu.Game/Overlays/Mods/ModPanel.cs

```diff
@@ -37,6 +37,8 @@ public ModPanel(ModState modState)
                 Shear = new Vector2(-ShearedOverlayContainer.SHEAR, 0),
                 Scale = new Vector2(HEIGHT / ModSwitchSmall.DEFAULT_SIZE)
             };
+
+            Action = select;
```

Comment:
I'm very much not a fan of this being a silent "override" of the base class' behaviour, based on it being set and overwriting the delegate set by `ModSelectPanel`..

Reply:
I wasn't expecting this to be a problem if I'm honest. Given that `ModPresetPanel` also overwrites the action, I may as well remove the set in the base class entirely, if it helps.

Reply:
Hmm, it's just not a standard way of doing things (and easy to miss). While reading through this change I was very confused how anything was working, until I realised that this flow was being overwritten.

A better way might be to expose a `virtual` method in the base class and use overriding of that rather than `Action=` directly.

Reply:
I've made it `abstract` instead, because as it turns out, if I did the "sensible" default in `ModSelectPanel` by setting `Active` to true/false directly, I would also have to bypass base calls in `ModPresetPanel` because of how that couples the active state to selected mods.